src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/FeaturizedReadSets.java

```diff
@@ -0,0 +1,148 @@
+                         final GenotypeBuilder gb,
+                         final AlleleLikelihoods<GATKRead, Allele> likelihoods,
+                         final AlleleLikelihoods<Fragment, Allele> fragmentLikelihoods,
+                         final AlleleLikelihoods<Fragment, Haplotype> haplotypeLikelihoods) {
+        Utils.nonNull(vc);
+        Utils.nonNull(g);
+        Utils.nonNull(gb);
+
+        if ( likelihoods == null) {
+            return;
+        }
+
+        final Map<Allele, List<GATKRead>> readsByAllele = likelihoods.alleles().stream()
+                .collect(Collectors.toMap(a -> a, a -> new ArrayList<>()));
+
+        Utils.stream(likelihoods.bestAllelesBreakingTies())
+                .filter(ba -> ba.isInformative())
+                .forEach(ba -> readsByAllele.get(ba.allele).add(ba.evidence));
+
+        // downsample if necessary
+        final Allele refAllele = likelihoods.alleles().stream().filter(Allele::isReference).findFirst().get();
+        if (readsByAllele.get(refAllele).size() > maxRefCount) {
+            Collections.shuffle(readsByAllele.get(refAllele));
+            readsByAllele.put(refAllele, readsByAllele.get(refAllele).subList(0, maxRefCount));
+        }
+
+        final Map<GATKRead, Haplotype> bestHaplotypes = new HashMap<>();
+        haplotypeLikelihoods.bestAllelesBreakingTies().stream().forEach(ba ->
+            ba.evidence.getReads().forEach(read -> bestHaplotypes.put(read, ba.allele)));
+
+        final List<String> stringsInAlleleOrder = vc.getAlleles().stream()
+                .map(allele -> {
+                            final List<GATKRead> reads = readsByAllele.get(allele);
+                            final List<Integer> flattened = new ArrayList<>(reads.size() * FEATURES_PER_READ);
+                            reads.forEach(read -> flattened.addAll(featurize(read, vc, bestHaplotypes)));
+                            return StringUtils.join(flattened, ",");
+                        }).collect(Collectors.toList());
+
+
+        final String annotation = AnnotationUtils.encodeAnyASListWithRawDelim(stringsInAlleleOrder);
+
+        gb.attribute(GATKVCFConstants.FEATURIZED_READ_SETS_KEY, annotation);
+    }
+
+
+
+    @Override
+    public List<String> getKeyNames() {
+        return Collections.singletonList(GATKVCFConstants.FEATURIZED_READ_SETS_KEY);
+    }
+
+    private List<Integer> featurize(final GATKRead read, final VariantContext vc, final Map<GATKRead, Haplotype> bestHaplotypes) {
+        final List<Integer> result = new ArrayList<>();
+        result.add(read.getMappingQuality());
+        result.add(BaseQuality.getBaseQuality(read, vc).orElse(DEFAULT_BASE_QUALITY));
+        result.add(read.isFirstOfPair() ? 1 : 0);
+        result.add(read.isReverseStrand() ? 1 : 0);
+
+        // distances from ends of read
+        final int readPosition = ReadPosition.getPosition(read, vc).orElse(0);
+        result.add(readPosition);
+        result.add(read.getLength() - readPosition);
```

Comment:
If this is the distance to either end of the read, `Math.min(readPosition, read.getLength() - readPosition)`

Reply:
We're annotating both distances here — it's not the same as the `ReadPosition` annotation.